backend/treebanks/views.py

```diff
@@ -1,16 +1,23 @@
 from .models import Treebank, Component
 from .serializers import TreebankSerializer, ComponentSerializer
-from rest_framework import viewsets
+from rest_framework import viewsets, status
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
 
 
-class TreebankViewset(viewsets.ModelViewSet):
+class TreebankViewset(viewsets.ReadOnlyModelViewSet):
     # TODO: make sure that non-public treesets are hidden if needed
     queryset = Treebank.objects.all().order_by('slug')
     serializer_class = TreebankSerializer
 
 
-class ComponentViewset(viewsets.ModelViewSet):
-    # TODO: make sure that components of non-public treebanks are hidden if
-    # needed
-    queryset = Component.objects.all().order_by('slug')
-    serializer_class = ComponentSerializer
+@api_view(['GET'])
+def components_for_treebank_view(request, treebank):
```

Comment:
I personally like function-based views, but if you'd rather use drf's classes, I think this should be possible using `ListAPIView` with a `get_queryset()` method

Reply:
I have changed the other view to a function-based view as well for consistency.